embedchain/vectordb/lancedb.py

```diff
@@ -0,0 +1,309 @@
+        :param metadatas: Metadatas
+        :type metadatas: List[object]
+        :param ids: ids
+        :type ids: List[str]
+        """
+        data = []
+        to_ingest = list(zip(documents, metadatas, ids))
+
+        if not self.embedder_check:
+            for doc, meta, id in to_ingest:
+                temp = {}
+                temp["doc"] = doc
+                temp["metadata"] = str(meta)
+                temp["id"] = id
+                data.append(temp)
+        else:
+            for doc, meta, id in to_ingest:
+                temp = {}
+                temp["doc"] = doc
+                temp["vector"] = self.embedder.embedding_fn([doc])[0]
+                temp["metadata"] = str(meta)
+                temp["id"] = id
+                data.append(temp)
+
+        self.collection.add(data=data)
+
+    def _format_result(self, results) -> list:
+        """
+        Format LanceDB results
+
+        :param results: LanceDB query results to format.
+        :type results: QueryResult
+        :return: Formatted results
+        :rtype: list[tuple[Document, float]]
+        """
+        return results.tolist()
+
+    def query(
+        self,
+        input_query: str,
+        n_results: int = 3,
+        where: Optional[dict[str, any]] = None,
+        raw_filter: Optional[dict[str, any]] = None,
+        citations: bool = False,
+        **kwargs: Optional[dict[str, any]],
+    ) -> Union[list[tuple[str, dict]], list[str]]:
+        """
+        Query contents from vector database based on vector similarity
+
+        :param input_query: query string
+        :type input_query: str
+        :param n_results: no of similar documents to fetch from database
+        :type n_results: int
+        :param where: to filter data
+        :type where: dict[str, Any]
+        :param raw_filter: Raw filter to apply
+        :type raw_filter: dict[str, Any]
+        :param citations: we use citations boolean param to return context along with the answer.
+        :type citations: bool, default is False.
+        :raises InvalidDimensionException: Dimensions do not match.
+        :return: The content of the document that matched your query,
+        along with url of the source and doc_id (if citations flag is true)
+        :rtype: list[str], if citations=False, otherwise list[tuple[str, str, str]]
+        """
+        if where and raw_filter:
+            raise ValueError("Both `where` and `raw_filter` cannot be used together.")
+        try:
+            query_embedding = self.embedder.embedding_fn(input_query)[0]
+            result = self.collection.search(query_embedding).limit(n_results).to_list()
+        except Exception as e:
+            e.message()
+            +". This is commonly a side-effect when an embedding function, different from the one used to add the embeddings, is used to retrieve an embedding from the database."  # noqa E501
+
+        # results_formatted = self._format_result(result)
```

Comment:
This line can be removed if it won't be used later

Reply:
ok